src/extension.ts

```diff
@@ -129,6 +129,7 @@ export class Extension implements RunHooks {
   }
 
   async onWillRunTests(config: TestConfig, debug: boolean) {
+    this._models.onWillRunTests();
```

Comment:
This one is a little funny because embedder (extension) notifies model collection that a model called the embedder and notified it about the runTests event. I would call collection.saveSettings right where the model runs tests instead to save on the indirection.

Reply:
It is! Didn't feel like I could just remove the `private`, so I added some indirection. But it looks like `TestModel` also calls `TestModelCollection#_modelUpdated`, so they trust one another. Made the change!